src/FactSystem/FactControls/FactStepper.qml

```diff
@@ -0,0 +1,130 @@
+
+import QtQuick 2.5
+import QtQuick.Controls 1.2
+import QtQuick.Controls.Styles 1.2
+import QtQuick.Dialogs         1.2
+
+import QGroundControl.ScreenTools   1.0
+import QGroundControl.FactSystem 1.0
+import QGroundControl.Palette 1.0
+import QGroundControl.Controls 1.0
+
+Item {
+    id: stepper
+    property bool enabled: true
+    property alias fact: factInput.fact
+    property alias showHelp: factInput.showHelp
+    property alias showUnits: factInput.showUnits
+
+    property string _valueText:          fact ? fact.valueString : "N/A"
+    property double _factValue:          Number(_valueText)
+
+    property double minimumValue:       !fact || isNaN(fact.min) ? 0 : fact.min
+    property double maximumValue:       !fact || isNaN(fact.max) ? 1 : fact.max
+    property double stepSize:           !fact || isNaN(fact.increment) ? 0 : fact.increment
+    property double stepRatio:          0.1
+
+    implicitHeight: 50
+    implicitWidth: implicitHeight * 4
+
+    signal validationError(string text)
+    signal helpClicked
+
+    Component.onCompleted: {
+        factInput.validationError.connect(validationError)
+        factInput.helpClicked.connect(helpClicked)
+   }
+
+    function incrementWithScale(multiplier) {
+        var step = stepSize;
+        if (step == 0) {
+            step = Math.abs(stepRatio * _factValue);
+        }
+
+        factInput.setFactValue(_factValue + (step * multiplier));
+    }
+
+    Timer {
+        id: factValueChangeDelay
+
+        interval: 250
+        repeat: false
+        running: false
+
+        onTriggered: {
+            fact.value = _factValue;
+            fact.valueChanged(fact.value)
+        }
+    }
+
+    Row {
+        anchors.fill:   parent
+
+        QGCButton {
+            id: minusButton
+            width : parent.width / 4
+            height: parent.height
+            text: "-"
+            enabled: _factValue > minimumValue
+
+            onClicked: {
+                incrementWithScale(-1)
+            }
+        }
+
+        FactTextField {
+            id: factInput
+            width : parent.width / 2
+            height: parent.height
+
+            horizontalAlignment: TextInput.AlignHCenter
+
+            text:       _valueText
+
+            fact: _fact
+
+            function setFactValue(newValue) {
+                if (newValue < fact.min) {
+                    newValue = fact.min;
+                }
+                if (newValue > fact.max) {
+                    newValue = fact.max;
+                }
+
+                factValueChangeDelay.stop()
+                _valueText = newValue.toLocaleString(Qt.locale(), 'f', fact.decimalPlaces)
+
+                console.log(fact.name, ": _factValue=", _factValue)
+                factValueChangeDelay.start()
+            }
+
+            onValidationError: {
+                // do nothing by default
+            }
+
+            onHelpClicked: {
+                // do nothing by default
+            }
+        }
```

Comment:
The idea of the factValueChangeDelay is to prevent to frequent updates of values, as it results on value set drop outs.
This is critical when WIFI or BT is used with QGC on mobile phone and user holds the +/- button to 'scroll' to needed value.